.github/workflows/packaging-test-pipelines.yml

```diff
@@ -0,0 +1,161 @@
+
+  build_tests_for_rpm_in_packaging_images:
+    name: tests_build_in_packaging_images
+    needs: get_postgres_versions_from_file
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        # While we use separate images for different Postgres versions in rpm
+        # based distros
+        # For this reason, we need to use a "matrix" to generate names of
+        # rpm images, e.g. citus/packaging:centos-7-pg12
+        packaging_docker_image:
+          - citus/packaging:oraclelinux-7-pg
+          - citus/packaging:oraclelinux-8-pg
+          - citus/packaging:centos-7-pg
+          - citus/packaging:centos-8-pg
+          - citus/packaging:almalinux-9-pg
+        POSTGRES_VERSION: ${{ fromJson(needs.get_postgres_versions_from_file.outputs.pg_versions) }}
+
+    container:
+      image: ${{ matrix.packaging_docker_image }}${{ matrix.POSTGRES_VERSION }}
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Add Postgres installation directory into PATH for rpm based distros
+        run: |
+          echo "/usr/pgsql-${{ matrix.POSTGRES_VERSION }}/bin" >> $GITHUB_PATH
+
+      - name: Configure
+        run: |
+          echo "Current Shell:$0"
+          echo "GCC Version: $(gcc --version)"
+          ./configure 2>&1 | tee output.log
+
+      - name: Make clean
+        run: |
+          make clean
+
+      - name: Make
+        run: |
+          make -sj$(cat /proc/cpuinfo | grep "core id" | wc -l) 2>&1 | tee -a output.log
+
+      - name: Make install
+        run: |
+          make install 2>&1 | tee -a output.log
+
+      - name: Validate output
+        env:
+          POSTGRES_VERSION: ${{ matrix.POSTGRES_VERSION }}
+          PACKAGING_DOCKER_IMAGE: ${{ matrix.packaging_docker_image }}
+        run: |
+          echo "Postgres version: ${POSTGRES_VERSION}"
+
+          package_type="rpm"
+          ## Install required packages to execute packaging tools for rpm based distros
+          yum install python3-pip python3-devel postgresql-devel -y
+          python3 -m pip install wheel
+
+          git clone -b citus_packaging_pipeline --depth=1  https://github.com/citusdata/tools.git tools
+          python3 -m pip install -r tools/packaging_automation/requirements.txt
+          python3 -m tools.packaging_automation.validate_build_output --output_file output.log \
+            --ignore_file .github/packaging/packaging_ignore.yml --package_type ${package_type}
+
+  build_tests_for_deb_in_packaging_images:
+    name: tests_build_in_packaging_images
+    needs: get_postgres_versions_from_file
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        #  We use the same docker image for builds based on
+        # different Postgres versions in deb based distros, since images includes all postgres installations.
+        # For this reason, we don't use POSTGRES_VERSION in container name but used it to set PG_CONFIG variable
```

Comment:
```suggestion
        # On deb based distros, we use the same docker image for
        # builds based on different Postgres versions because deb
        # based images include all postgres installations.
        # For this reason, we have multiple runs --which is 3 today--
        # for each deb based image and we use POSTGRES_VERSION to set
        # PG_CONFIG variable in each of those runs.
```